flink-core/src/main/java/org/apache/flink/api/common/TaskRuntimeInfo.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common;
+
+/**
+ * Encapsulation of runtime information for a Task, including Task Name, sub-task index, etc.
+ *
+ */
+public class TaskRuntimeInfo implements java.io.Serializable {
+
+	/** Task Name */
+	private final String taskName;
+
+	/** Index of this task in the parallel task group. Goes from 0 to {@link #numParallelTasks} - 1 */
+	private final int subTaskIndex;
+
+	/** Number of parallel running instances of this task */
+	private final int numParallelTasks;
+
+	/** Attempt number of this task */
+	private final int attemptNumber;
+
+	/** Task Name along with information on index of task in group */
+	private transient final String taskNameWithSubTaskIndex;
```

Comment:
If this is transient, it is null after deserialization. Either make it non-transient, or re-compute it after deserialization.